api/controllers/UserController.js

```diff
@@ -1593,23 +1884,32 @@ module.exports = {
         }
       }
       const promises = [];
-      const pendingLogs = [];
-      promises.push(record.save());
-      pendingLogs.push({
-        type: 'info',
-        message: `[UserController->validateEmail] Saved user ${record.id} successfully`,
-      });
+      promises.push(record.save().then(() => {
```

Comment:
Not related to this PR but I have a concern here. The calls to `save`, `sendPostRegister` are not synchronized and `sendPostRegister` is not dependent of the success of `save`. The post_register email may be sent even if saving the user record fails.

Reply:
haha dope. Great catch and i'll ticket the issue.

We already have the mechanism in place (node.js `async`/`await`) so we'll make these wait on each other and get rid of the whole promise queue. There's backstory here that I don't want to retell so I'll just say I'm happy to make this change and use `async`/`await` as intended.

Reply:
HID-2102